src/Sequence.cc

```diff
@@ -635,9 +635,11 @@ void Sequence::set_step_setup_script(gul14::string_view step_setup_script)
     step_setup_script_.assign(step_setup_script.data(), step_setup_script.size());
 }
 
-void Sequence::set_error_message(gul14::string_view msg)
+void Sequence::set_error_message_and_step_index(gul14::string_view msg,
+    OptionalStepIndex optional_step_index)
 {
     error_message_.assign(msg.data(), msg.size());
+    error_step_index_ = optional_step_index;
```

Comment:
I believe this should raise some kind of error if the string is empty (i.e. _no error_ ? and the optional_step_index is not empty.
At least it should be mentioned in the Doxygen of this function that 'this run had errors' is transported by the filling of the string and only that. That it is not allowed to have an empty error message on index 42. This feels trivial and obvious at the moment, but I bet users will wonder... I see me as user grepping the code to find out.

Reply:
This would be solved by the new proposal I guess.